### ISSUE TITLE
torch.quantization.fuse_modules behavior different than origin model.

### ISSUE THREAD

**Issue by jinfagang** [pytorch/pytorch#52584]
```python


model = resnet18(num_classes=num_classes, pretrained=False)
model.to(cpu_device)
print(model)
# Make a copy of the model for layer fusion
fused_model = copy.deepcopy(model)

model.train()
# The model has to be switched to training mode before any layer fusion.
# Otherwise the quantization aware training will not work correctly.
fused_model.train()

# Fuse the model in place rather manually.
fused_model = torch.quantization.fuse_modules(
    fused_model, [["conv1", "bn1", "relu"]], inplace=True)
for module_name, module in fused_model.named_children():
    if "layer" in module_name:
        for basic_block_name, basic_block in module.named_children():
            torch.quantization.fuse_modules(
                basic_block, [["conv1", "bn1", "relu"], ["conv2", "bn2"]], inplace=True)
            for sub_block_name, sub_block in basic_block.named_children():
                if sub_block_name == "downsample":
                    torch.quantization.fuse_modules(
                        sub_block, [["0", "1"]], inplace=True)
```

Simply model., fuse using `torch.quantization` the result not same:

```
def model_equivalence(model_1, model_2, device, rtol=1e-05, atol=1e-08, num_tests=100, input_size=(1, 3, 32, 32)):
    model_1.to(device)
    model_2.to(device)

    for _ in range(num_tests):
        x = torch.rand(size=input_size).to(device)
        y1 = model_1(x).detach().cpu().numpy()
        y2 = model_2(x).detach().cpu().numpy()
        if np.allclose(a=y1, b=y2, rtol=rtol, atol=atol, equal_nan=False) == False:
            print("Model equivalence test sample failed: ")
            print(y1)
            print(y2)
            return False
    return True

assert model_equivalence(model_1=model, model_2=fused_model, device=cpu_device, rtol=1e-03, atol=1e-06,
                             num_tests=100, input_size=(1, 3, 32, 32)), "Fused model is not equivalent to the original model!"
```

Anybody knows what goes wrong?

cc @jerryzh168 @jianyuh @dzhulgakov @raghuramank100 @jamesr66a @vkuzo

**Comment by vkuzo** [comment]
hi @jinfagang , could you try doing this in eval mode?  If you compare in train mode, then you would be using the current batch's training stats in the unfused model, and the stats used during fusion in the fused model.

**Comment by EchoIR** [comment]
@vkuzo hello, I try doing this in eval mode, but i got different results between the origin model and the fused model, WHY????

**Comment by vkuzo** [comment]
hi @EchoIR , please post your question on https://discuss.pytorch.org/c/quantization/17 with a repro, that would make it easier for folks to help out.

**Comment by EchoIR** [comment]
好了，折叠有问题~感谢 @vkuzo

**Comment by kofi2614** [comment]
Hi @jinfagang, I'm looking at the exact same code and having the same issue. Have you resolved this? Do you know what's the reason of mismatch?

**Comment by jinfagang** [comment]
I am using torch.fx to convert model for quantization, it works perfectly.
